### PR TITLE
Added CommonPythonTest.cmake to add Python tests to a project.

### PR DESCRIPTION
**Pull request by hernando**
This script works in a similar way to CommonCPPCTest

### CODE REVIEW

**Comment by hernando** [comment]
@tribal-tec please review

File: CommonPythonCTest.cmake [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2016, Juan Hernando <juan.hernando@fi.upm.es>
+#
+# Creates a per project pythontests target.
+# This target creates a test for each .py file in the current
+# directory tree. The per project targett is included into the global
```
Comment (tribal-tec) [inline CommonPythonCTest.cmake]: typo targett

File: CommonPythonCTest.cmake [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# 'tests' from CommonCTest.cmake.
+#
+# Input:
+# * PYTHON_TEST_DEPENDENCIES targets that are required by the tests
+# * EXCLUDE_FROM_TESTS relative paths to test files to exclude; optional
```
Comment (tribal-tec) [inline CommonPythonCTest.cmake]: PYTHON_TEST_EXCLUDE for consistency in this file at least? We have to touch the CPP one as well at some point...

File: CommonPythonCTest.cmake [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    COMMENT "Running ${TEST_NAME} python test")
+  set_target_properties(ctest_${TEST_NAME} PROPERTIES
+    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "Tests")
+
+  set(ALL_PYTHON_TESTS ${ALL_PYTHON_TESTS} ctest_${TEST_NAME} PARENT_SCOPE)
+endfunction()
+
+foreach(TEST ${PYTHON_TEST_EXECUTABLES})
+  common_add_python_test(${TEST})
+endforeach(TEST)
```
Comment (tribal-tec) [inline CommonPythonCTest.cmake]: remove TEST within parentheses

**Comment by tribal-tec** [comment]
+1 otherwise

**Comment by hernando** [comment]
I've realized that this code doesn't handle changes in the test source files. To fix that it seems it's better to refactor the whole code.
The EXCLUDE_FROM_TEST variable was not consider either.

**Comment by hernando** [comment]
All fixed.

**Pull request merged**